sdk/containerregistry/container-registry/src/containerRepository.ts

```diff
@@ -0,0 +1,346 @@
+ * @internal
+ */
+export class ContainerRepositoryImpl {
+  private readonly client: GeneratedClient;
+  /**
+   * The Azure Container Registry endpoint.
+   */
+  public readonly registryUrl: string;
+  /**
+   * Repository name.
+   */
+  public readonly name: string;
+  /**
+   * Registry name.
+   */
+  public readonly fullyQualifiedName: string;
+
+  /**
+   * Creates an instance of a ContainerRepository.
+   * @param registryUrl - the URL to the Container Registry endpoint
+   * @param name - the name of the repository
+   * @param client - the generated client that interacts with service
+   */
+  constructor(registryUrl: string, name: string, client: GeneratedClient) {
+    this.registryUrl = registryUrl;
+    this.name = name;
+    const parsedUrl = new URL(registryUrl);
+    this.fullyQualifiedName = `${parsedUrl.hostname}/${name}`;
+
+    this.client = client;
+  }
+
+  /**
+   * Deletes this repository.
+   *
+   * @param options - optional configuration for the operation
+   */
+  public async delete(options: DeleteRepositoryOptions = {}): Promise<DeleteRepositoryResult> {
+    const { span, updatedOptions } = createSpan("ContainerRepository-delete", options);
+
+    try {
+      const result = await this.client.containerRegistry.deleteRepository(
+        this.name,
+        updatedOptions
+      );
+      return {
+        deletedManifests: result.deletedManifests ?? [],
+        deletedTags: result.deletedTags ?? []
+      };
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Returns an instance of RegistryArtifact.
+   * @param tagOrDigest - the tag or digest of the artifact
+   */
+  public getArtifact(tagOrDigest: string): RegistryArtifact {
+    return new RegistryArtifactImpl(this.registryUrl, this.name, tagOrDigest, this.client);
+  }
+
+  /**
+   * Retrieves properties of this repository.
+   * @param options -
+   */
+  public async getProperties(
+    options: GetRepositoryPropertiesOptions = {}
+  ): Promise<RepositoryProperties> {
+    const { span, updatedOptions } = createSpan("ContainerRepository-getProperties", options);
+
+    try {
+      const result = await this.client.containerRegistry.getProperties(this.name, updatedOptions);
+      return result;
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Updates repository attributes.
+   * @param options -
+   */
+  public async setProperties(
+    options: SetRepositoryPropertiesOptions
+  ): Promise<RepositoryProperties> {
+    const { span, updatedOptions } = createSpan("ContainerRepository-setProperties", {
+      ...options,
+      value: {
+        canDelete: options.canDelete,
+        canWrite: options.canWrite,
+        canList: options.canList,
+        canRead: options.canRead
+      }
```

Comment:
Is there any reason to pass these individually rather than trusting the `...value` above?

Reply:
Gotcha! Thank you